doc/src/manual/types.md

```diff
@@ -1002,6 +1002,20 @@ For example `Tuple{T,Array{S}} where S<:AbstractArray{T} where T<:Real` refers t
 element is some `Real`, and whose second element is an `Array` of any kind of array whose element type
 contains the type of the first tuple element.
 
+The `where` keyword itself can be nested inside a more complex declaration.  For example, consider the
+two types created by the following declarations:
+```julia
+julia> const T1 = Array{Array{T,1} where T, 1}
+Array{Array{T,1} where T,1}
+
+julia> const T2 = Array{Array{T,1}, 1} where T
+Array{Array{T,1},1} where T
+```
+Type `T1` defines a 1-dimensional array of 1-dimensional arrays; each
+of the inner arrays consists of objects of the same type, but this type may vary from one inner array to the next.
+On the other hand, type `T2` defines a 1-dimensional array of 1-dimensional arrays all of whose inner arrays must have the 
```

Comment:
I will just state my opinion that trailing whitespace in the documentation is an absurd reason for a build to fail.